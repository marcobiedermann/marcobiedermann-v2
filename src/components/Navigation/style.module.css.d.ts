/* eslint-disable import/prefer-default-export */

export const navigation: string;
