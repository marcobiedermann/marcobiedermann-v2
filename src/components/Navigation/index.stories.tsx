/* eslint-disable import/no-extraneous-dependencies */

import { storiesOf } from '@storybook/react';
import React from 'react';
import Navigation from '.';

storiesOf('Navigation', module).add('default', () => <Navigation routes={[]} />);
