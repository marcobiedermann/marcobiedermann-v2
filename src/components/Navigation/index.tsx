import classNames from 'classnames';
import React, { FC } from 'react';
import Route, { RouteProps } from '../Route';
import styles from './style.module.css';

export interface NavigationProps {
  className?: string;
  routes: RouteProps[];
}

export const Navigation: FC<NavigationProps> = (props) => {
  const { className, routes, ...otherProps } = props;

  return (
    <nav className={classNames(className, styles.navigation)} {...otherProps}>
      <ul className={styles.navigation__list}>
        {routes.map((route) => (
          <li key={route.id}>
            <Route activeClassName={styles.navigation__linkActive} className={styles.navigation__link} {...route} />
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navigation;
