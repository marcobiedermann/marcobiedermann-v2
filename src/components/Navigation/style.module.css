.navigation {
  text-transform: uppercase;
}

.navigation__link {
  text-decoration: none;
}

.navigation__link--active {
  font-weight: bold;
}
