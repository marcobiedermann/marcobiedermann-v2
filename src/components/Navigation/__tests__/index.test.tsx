/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Navigation from '..';

describe('Navigation component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Navigation routes={[]} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
