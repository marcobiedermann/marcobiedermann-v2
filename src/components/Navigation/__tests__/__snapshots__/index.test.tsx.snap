// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should render correctly 1`] = `
<DocumentFragment>
  <nav
    class="navigation"
  >
    <ul
      class="navigation__list"
    />
  </nav>
</DocumentFragment>
`;
